soc/espressif/Kconfig.ulp

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2025 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+menu "Ultra Low Power (ULP) Co-processor"
```

Comment:
Would be good to use either `Coprocessor` or `Co-processor` along the PR/documentation. I've seen both being used.